webapp/src/components/call_error_modal.tsx

```diff
@@ -0,0 +1,152 @@
+
+    const onRejoinClick = (ev: React.MouseEvent) => {
+        ev.preventDefault();
+        window.postMessage({type: 'connectCall', channelID: clientErr.channelID}, window.origin);
+        dispatch(clearClientError());
+    };
+
+    const onConfirm = () => {
+        if (clientErr.err === insecureContextErr) {
+            window.open('https://docs.mattermost.com/configure/calls-deployment.html', '_blank');
+        }
+        return null;
+    };
+
+    const troubleShootingMsg = (
+        <React.Fragment>
+            {' Check the '}
+            <a href='https://docs.mattermost.com/channels/make-calls.html#troubleshooting'>{'troubleshooting section'}</a>
+            {' if the problem persists.'}
+        </React.Fragment>
+    );
+
+    const genericMsg = (
+        <span>
+            {'Looks like something went wrong with calls. You can restart the app and try again.'}
+            {troubleShootingMsg}
+        </span>
+    );
+    const genericHeaderMsg = (
+        <span>
+            {'Something went wrong with calls'}
+        </span>
+    );
+
+    let msg = genericMsg;
+    let headerMsg = genericHeaderMsg;
+    let confirmMsg = 'Okay';
+
+    switch (clientErr.err) {
+    case rtcPeerErr:
+    case rtcPeerCloseErr:
+        headerMsg = (
+            <span>{'Connection failed'}</span>
+        );
+        msg = (
+            <span>
+                {'There was an error with the connection to the call. Try to '}
+                <a
+                    href=''
+                    onClick={onRejoinClick}
+                >{'re-join'}</a>
+                {' the call.'}
+                {troubleShootingMsg}
+            </span>
+        );
+        break;
+    case insecureContextErr:
+        headerMsg = (
+            <ColumnContainer>
+                <LaptopAlertSVG
+                    width={150}
+                    height={150}
+                />
+                <span>{'Cannot initialize calls in an insecure context'}</span>
+            </ColumnContainer>
+        );
+        msg = (
+            <span>
+                {'Looks like you are not on a secure connection. Calls require an HTTPs connection to work. Check out the documentation for more information.'}
+            </span>
+        );
+        modalProps.showCancel = true;
+        modalProps.cancelButtonText = 'Cancel';
+        confirmMsg = 'Learn more';
+        break;
+    }
+
+    return (
+        <StyledGenericModal
+            {...modalProps}
+            id={CallErrorModalID}
+            modalHeaderText={headerMsg}
+            confirmButtonText={confirmMsg}
+            onHide={() => dispatch(clearClientError())}
+            handleConfirm={onConfirm}
+            contentPadding={'48px 32px'}
+        >
+            <ColumnContainer>
+                {msg}
+            </ColumnContainer>
+        </StyledGenericModal>
+    );
+};
+
+const StyledGenericModal = styled(GenericModal)`
+  width: 512px;
+
+  &&& {
+    .modal-header, .modal-footer {
```

Comment:
The generic modal allows you to pass in the header and footer in the `components` props, so that we don't have to do this css hacking.